src/services/db/GitFileSystemService.ts

```diff
@@ -473,8 +499,78 @@ export default class GitFileSystemService {
       )
   }
 
-  // TODO: Update the contents of a file
-  async update() {}
+  // Update the contents of a file
+  update(
+    repoName: string,
+    filePath: string,
+    fileContent: string,
+    oldSha: string,
+    userId: SessionDataProps["isomerUserId"]
+  ): ResultAsync<string, GitFileSystemError | NotFoundError | ConflictError> {
+    return ResultAsync.fromPromise(
+      fs.promises.stat(`${EFS_VOL_PATH}/${repoName}/${filePath}`),
+      (error) => {
+        if (error instanceof Error && error.message.includes("ENOENT")) {
```

Comment:
I find repeated checks for this ENOENT and not found error, may can abstract out to a util that checks

Reply:
Good point, done in 49aff43f0a98e202aab77ae65dd8d64a103d26b6.